
# Introdution {.unnumbered}

**Objectives** : 

Yesterday, we saw how to pre-process single cell rna sequencing data (Drop-seq) to extract "high quality cells". Today, we will explore further a similar dataset and use a computional approach called _novoSparc_ to reconstruct the spatial transcriptome. Before performing an advanced technic such as spatial transcriptomics, it is good to have some kind of validation. So as the first step, we will visualize co-expressed genes in UMAP and see if we can find the same co-expression in our predicted spatial transcriptome. 


In this pratical session, you will learn : 

* how to visualize co-expressed genes in UMAP ;
* how to map dissociated single cells back to their tissue of origin using novoSpaRc (de novo Spatial Reconstruction of Single-Cell Gene Expression).

We will be using one of the dataset publilshied with Nitizan et al (Nature 2019). The data are alreay filtered ("high quality cells") and log-normalized. 


## Sources of inspiration{.unnumbered}

This course material is based on Novosparc tutorial on github:

* [novosparc tutorial](https://github.com/rajewsky-lab/novosparc): reconstruct_drosophila_embryo_tutorial.ipynb

## Bibliography{.unnumbered}

* Karaiskos et al.The Drosophila embryo at single-cell transcriptome resolution. Science13 Oct 2017 : 194-199 
* Nitzan  et al. Gene expression cartography. Nature.2019 Dec;576(7785):132-137.  
* Tutorial [working with Python in R](https://bookdown.org/yihui/rmarkdown-cookbook/eng-python.html)

# What to install
If you work from your local computer, you need :

* Install the following R library using the console :
    + install.package("devtool") : to get the `install_github` function
    + install_github("rstudio/reticulate") : to install the dev version of 
    reticulate and work with Python in R
    + install.package("bookdown") (to have a cool notebook at the end)
    + install.package("knitr") : to manange the outputs (figures..)
    + install.package("Seurat ") : useful to analyse scRNAseq data in R
    
* Install in Python3 using "Terminal"
    + pip3 install numpy : to manipulate data arry
    + pip3 install matplotlib : for graphs
    + pip3 install novosparc : for sc mapping
    + pip3 install anndata : : useful to analyse scRNAseq data in Python

# Working in Python in RStudio
RStudio makes it easy to combine R and Python in a single data science project thanks to the package "reticulate". You just need to specify the language you want to use at the beginning of the chunk.


```{r setup}
library(reticulate) # to use Python through a R framework
library(knitr)
knitr::knit_engines$set(python=reticulate::eng_python)
# optional : check the version of Python that you are using (not useful for this workshop)
# use_python("/usr/local/bin/python3.6", required = TRUE) 
```
To import package from Python, the syntax is a bit different:

```{python3, message = FALSE}
import numpy as np   # we use the alias np to call function from the package numpy
#import novosparc
import matplotlib.pyplot as plt
#import anndata as an
```

Create a variable `x` in  Python :
```{python3, message=FALSE}
x = np.array([1,2,3])
print(x)
```
Use the variable x in R by adding *py$* in front of the variable name
```{r, message=FALSE}
py$x
```
You can also create a new variable `y` in the Python session using R, and pass a data frame to `y`:

```{r, message=FALSE}
py$y <- head(cars) # cars is a dataset which is pre-installed in R
```
Print the variable `y` in Python:
```{python3, message=FALSE, message = FALSE}
print(y)
```
# Basic vizualization
In this section, we will perfome some basic analyses of the dataset provided in Nitizan et al. using Seurat. We will visualize the cells in a reduced space using UMAP and visualize the expression of some genes. 
If you are not familiar with Seurat object and scRNAseq pre-processing, you can look at yesterday's tutorial, which is available on [github](https://github.com/PSL-winter-school-2021/01-monday-workshop-rna). 

## Import data in Seurat 

```{r, echo=FALSE, eval=FALSE}
# install package if necessary
if (!requireNamespace("BiocManager", quietly = TRUE))
	    install.packages("BiocManager")
BiocManager::install("org.Dm.eg.db")
BiocManager::install("clusterProfiler")
```

```{r, message=FALSE, message = FALSE}
library(Seurat)
library(ggplot2)
library(org.Dm.eg.db)
library(clusterProfiler)
```

We import the log-normalized high quality cells as a dataframe and then convert the dataframe to a Seurat object. 
```{r, message=FALSE, message = FALSE}
log.nrmlz.matrix <- read.table("data/data_day2/drosophila_scRNAseq/dge_normalized.txt", sep='\t', row.names=1)
dim(log.nrmlz.matrix)
print(log.nrmlz.matrix[1:3,1:4])
```
We have 1963 genes and 1297 cells
```{r create-seurat-obj}
dataset <- CreateSeuratObject(counts =  log.nrmlz.matrix)
```
## A quick reminder on SeuratObject
In Seurat, data are organised in different compartements (slots), which contains them-selves several compartements, which can them-selves contain sub compartments...
```{r view-slotNames}
slotNames(dataset)
```

Each compartment can be used to store: 

* data from multiple modalities, such as RNAseq (slot `assays`, sub-slot `RNA`), ATAC-seq… 
* results of analyses: dimension reduction, clustering ...
* etc.

You navigate through this hierarchy using `@` and `$` signs.
```{r view-slotNames-assays}
slotNames(dataset@assays$RNA)
```

If you want to have a closer look at the data structure of the Seurat object, the R command `str()` might be very helpful:
```{r}
str(dataset)
```


The structure of the Seurat object also reflects the pre-processing steps. The scRNAseq workflow consists of 3 main steps :

* start with raw expression matrix (UMIs after sequencing) and filter out "low quality cells" : data are stored in the compartement `counts`; 
* log-normalize the data: to correct for sequencing biais : `data`
* perform a dimension reduction such as PCA to reduce the computation time and unnecessary information. PCA requires the data to be normalized and scaled : `scale.data`.

## Perform dimension reductions
When analysing scRNAseq data, you will usually perform some clustering to identify sub-populations and use some technique for visualization (e.g. UMAP).
These analyses are performed on a lower dimensional space.
Here we perform a PCA on the list of highly variable genes provided by the authors.

```{r load-hvg}
list_hvg  = read.csv("data/data_day2/drosophila_scRNAseq/high_var_genes.txt", header=FALSE)
```

```{r run-pca, warning=FALSE, message=FALSE}
dataset <- ScaleData(dataset)
dataset <- RunPCA(dataset, features = list_hvg$V1)
```

```{r plot-pca}
DimPlot(dataset, reduction = "pca")
sum(dataset@reductions$pca@stdev[1:35])
```

```{r plot-umap}
nPC          = 35 # number of PC kept for the analysis
n.neighbours = 15
resol        = 0.4 # impact the number of clusters 
# k-nn graph
dataset <- FindNeighbors(dataset, k.param = n.neighbours,dims = 1:nPC) 
# make the clusters
dataset <- FindClusters(dataset, resolution = resol) 
dataset <- RunUMAP(dataset, dims = 1:nPC)
DimPlot(dataset, reduction = "umap")
```

With the function `FeaturePlot` you can visualize the expression of some gene of interest on th UMAP:
```{r plot-genes}
FeaturePlot(object = dataset,
            features = c('twi','zen','aay', 'Antp'),
            cols = c("grey", "blue"),
            reduction = "umap",
            pt.size = 0.2)
```

You can define here two genes that you wish to see of the next plots (so you don't have to change the gene names in the plot functions).
```{r set-gene-names}
gene1 <- 'twi'
gene2 <- 'aay'
```

Let's have a look at the correlation between these two genes:
```{r plot-co-expression}
FeatureScatter(object = dataset, feature1 = gene1, feature2 = gene2)
```


```{r plot-genes-double}
p <- FeaturePlot(object = dataset,
         features = c(gene1, gene2),
         cols = c('grey', 'red', 'blue'),
         reduction = "umap",
         order = TRUE,
         pt.size = 0.5,
         combine = FALSE,
         blend = TRUE)
print(p[[3]] + NoLegend())

pp <- FeaturePlot(object = dataset,
         features = c(gene1, gene2),
         cols = c('white', 'red', 'blue'),
         reduction = "umap",
         order = TRUE,
         pt.size = 0.5,
         combine = FALSE,
         blend = TRUE)
print(pp[[3]] + NoLegend())
```



# Group discussion
* How could you identify the genes specifically expressed in one cluster ?
* Play with the colors of the plots. How much can this impact the resulting UMAP ?
* Try to find genes that are co-expressed (or anti-correlated) in this dataset.


## Interpretation PC axes
### Link with nUMI and nFeature

```{r prepare-to-plot-pca}
# Add PCs coordinates as a metadata
dataset <- AddMetaData(dataset, dataset@reductions$pca@cell.embeddings, colnames(dataset@reductions$pca@cell.embeddings))
# Add UMAP coordinates as a metadata
dataset <- AddMetaData(dataset, dataset@reductions$umap@cell.embeddings, colnames(dataset@reductions$umap@cell.embeddings))
# Check you have the PC aggregated to the metadata
head(dataset[[]])[1:10]
```

```{r plot-seqDepth}
FeaturePlot(object = dataset, features = "nCount_RNA",reduction = "pca")
FeaturePlot(object = dataset, features = "nFeature_RNA",reduction = "pca")

ggplot(dataset[[]],
	   aes(x=PC_1, y=PC_2, color = seurat_clusters)) +
  geom_point(aes(size = nCount_RNA)) +
  geom_rug()
  
ggplot(dataset[[]],
	   aes(x=UMAP_1, y=UMAP_2, color = seurat_clusters)) +
  geom_point(aes(size = nCount_RNA)) +
  geom_rug()
```

```{r select-pc}
# Extract PC gene loadings for the 10 first components
PC_df <- dataset@reductions$pca@feature.loadings[,1:10]
```

```{r select-loadings}
# Select PC to work with
nPC <- 1
# Select highest 5 genes in PC1
list_topVarPC1_plus <- sort(PC_df[,paste0('PC_', nPC)], decreasing = TRUE)[1:5]
list_topVarPC1_minus <- sort(PC_df[,paste0('PC_', nPC)], decreasing = FALSE)[1:5]
list_topVarPC1_plus
list_topVarPC1_minus
```



### Enrichement
```{r, eval=FALSE}
CallGroupGo = function(list_genes){
# groupGO return the GO profile at a specific level·
eg = bitr(list_genes, fromType="SYMBOL", toType=c("FLYBASE","ENTREZID"), OrgDb="org.Dm.eg.db")
# copy to clipboard for comparison with online app
#write_clip(list_topVarPC1)
ggo <- groupGO(gene     = na.omit(eg$ENTREZID),
               OrgDb    = org.Dm.eg.db,
               ont      = "BP",
               level    = 3,
               readable = TRUE)
return(ggo)
}

CallEnrichGo = function(list_genes,pval,qval){
#return the enrichment GO categories after FDR control.·
eg = bitr(list_genes, fromType="SYMBOL", toType=c("FLYBASE","ENTREZID"), OrgDb="org.Dm.eg.db")

ego <- enrichGO(gene          = na.omit(eg$ENTREZID),
                OrgDb         = org.Dm.eg.db,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = pval,
                qvalueCutoff  = qval,
        readable      = TRUE)
return(ego)
}

CallKeggPath = function(list_genes){
#return the enrichment GO categories after FDR control.·
eg = bitr(list_genes, fromType="SYMBOL", toType=c("FLYBASE","ENTREZID"), OrgDb="org.Dm.eg.db")

kk<- enrichKEGG(gene          = na.omit(eg$ENTREZID),
                organism      = org.Dm.eg.db,
                pvalueCutoff  = 0.01,
                use_internal_data = TRUE)
return(kk)
}
```

```{r enrich, eval=FALSE}
ego1p = CallEnrichGo(list_topVarPC1_plus,0.01,0.05)
ego1m = CallEnrichGo(list_topVarPC1_minus,0.01,0.05)

ego2 = CallEnrichGo(list_topVarPC2,0.02,0.15)
ego2p = CallEnrichGo(list_topVarPC2_plus,0.01,0.05)
ego2m = CallEnrichGo(list_topVarPC2_minus,0.01,0.05)

ego3 = CallEnrichGo(list_topVarPC3,0.02,0.15)
ego3p = CallEnrichGo(list_topVarPC3_plus,0.01,0.05)
ego3m = CallEnrichGo(list_topVarPC3_minus,0.01,0.05)
```

```{r, eval=FALSE}
#barplot(ggo1, drop=TRUE, showCategory=10,main = "Genes associated with PC1")
barplot(ego1, showCategory=25, main ="enrich genes PC1")
barplot(ego1p, showCategory=25, main ="enrich genes PC1")
barplot(ego1m, showCategory=25, main ="enrich genes PC1")
```
