<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 3 Basic vizualization | Mapping single-cell RNA-sequencing data on tissue using computations</title>
  <meta name="description" content=" 3 Basic vizualization | Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content=" 3 Basic vizualization | Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="git@github.com:PSL-winter-school-2021/02-tuesday-workshop-rna.git" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 3 Basic vizualization | Mapping single-cell RNA-sequencing data on tissue using computations" />
  
  
  

<meta name="author" content="Lorette Noiret, Nathalie Lehmann, Eric van Leen" />


<meta name="date" content="2021-02-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="working-in-python-in-rstudio.html"/>
<link rel="next" href="group-discussion.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="data/notebook/css/custom.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html"><i class="fa fa-check"></i>Introdution</a><ul>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#sources-of-inspiration"><i class="fa fa-check"></i>Sources of inspiration</a></li>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-to-install.html"><a href="what-to-install.html"><i class="fa fa-check"></i><b>1</b> What to install</a></li>
<li class="chapter" data-level="2" data-path="working-in-python-in-rstudio.html"><a href="working-in-python-in-rstudio.html"><i class="fa fa-check"></i><b>2</b> Working in Python in RStudio</a></li>
<li class="chapter" data-level="3" data-path="basic-vizualization.html"><a href="basic-vizualization.html"><i class="fa fa-check"></i><b>3</b> Basic vizualization</a><ul>
<li class="chapter" data-level="3.1" data-path="basic-vizualization.html"><a href="basic-vizualization.html#import-data-in-seurat"><i class="fa fa-check"></i><b>3.1</b> Import data in Seurat</a></li>
<li class="chapter" data-level="3.2" data-path="basic-vizualization.html"><a href="basic-vizualization.html#a-quick-reminder-on-seuratobject"><i class="fa fa-check"></i><b>3.2</b> A quick reminder on SeuratObject</a></li>
<li class="chapter" data-level="3.3" data-path="basic-vizualization.html"><a href="basic-vizualization.html#perform-dimension-reductions"><i class="fa fa-check"></i><b>3.3</b> Perform dimension reductions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="group-discussion.html"><a href="group-discussion.html"><i class="fa fa-check"></i><b>4</b> Group discussion</a><ul>
<li class="chapter" data-level="4.1" data-path="group-discussion.html"><a href="group-discussion.html#interpretation-pc-axes"><i class="fa fa-check"></i><b>4.1</b> Interpretation PC axes</a><ul>
<li class="chapter" data-level="4.1.1" data-path="group-discussion.html"><a href="group-discussion.html#link-with-numi-and-nfeature"><i class="fa fa-check"></i><b>4.1.1</b> Link with nUMI and nFeature</a></li>
<li class="chapter" data-level="4.1.2" data-path="group-discussion.html"><a href="group-discussion.html#enrichement"><i class="fa fa-check"></i><b>4.1.2</b> Enrichement</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>5</b> Session info</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Mapping single-cell RNA-sequencing data on tissue using computations</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="basic-vizualization" class="section level1">
<h1><span class="header-section-number"> 3</span> Basic vizualization</h1>
<p>In this section, we will perfome some basic analyses of the dataset provided in Nitizan et al. using Seurat. We will visualize the cells in a reduced space using UMAP and visualize the expression of some genes.
If you are not familiar with Seurat object and scRNAseq pre-processing, you can look at yesterday’s tutorial, which is available on <a href="https://github.com/PSL-winter-school-2021/01-monday-workshop-rna">github</a>.</p>
<div id="import-data-in-seurat" class="section level2">
<h2><span class="header-section-number">3.1</span> Import data in Seurat</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="basic-vizualization.html#cb8-1"></a><span class="kw">library</span>(Seurat)</span>
<span id="cb8-2"><a href="basic-vizualization.html#cb8-2"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb8-3"><a href="basic-vizualization.html#cb8-3"></a><span class="kw">library</span>(org.Dm.eg.db)</span>
<span id="cb8-4"><a href="basic-vizualization.html#cb8-4"></a><span class="kw">library</span>(clusterProfiler)</span></code></pre></div>
<p>We import the log-normalized high quality cells as a dataframe and then convert the dataframe to a Seurat object.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="basic-vizualization.html#cb9-1"></a>log.nrmlz.matrix &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/data_day2/drosophila_scRNAseq/dge_normalized.txt&quot;</span>, <span class="dt">sep=</span><span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>, <span class="dt">row.names=</span><span class="dv">1</span>)</span>
<span id="cb9-2"><a href="basic-vizualization.html#cb9-2"></a><span class="kw">dim</span>(log.nrmlz.matrix)</span></code></pre></div>
<pre><code>## [1] 1963 1297</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="basic-vizualization.html#cb11-1"></a><span class="kw">print</span>(log.nrmlz.matrix[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>])</span></code></pre></div>
<pre><code>##               GTACTAATTACN_2 AGGCTAATGGAC_3 TAGACAAAAGCT_4 TCTACTAGTGTN_5
## 128up                  2.570          2.509          0.000          0.000
## 140up                  1.160          3.007          0.000          0.000
## 14-3-3epsilon          8.838          8.788          8.843          8.869</code></pre>
<p>We have 1963 genes and 1297 cells</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="basic-vizualization.html#cb13-1"></a>dataset &lt;-<span class="st"> </span><span class="kw">CreateSeuratObject</span>(<span class="dt">counts =</span>  log.nrmlz.matrix)</span></code></pre></div>
</div>
<div id="a-quick-reminder-on-seuratobject" class="section level2">
<h2><span class="header-section-number">3.2</span> A quick reminder on SeuratObject</h2>
<p>In Seurat, data are organised in different compartements (slots), which contains them-selves several compartements, which can them-selves contain sub compartments…</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="basic-vizualization.html#cb14-1"></a><span class="kw">slotNames</span>(dataset)</span></code></pre></div>
<pre><code>##  [1] &quot;assays&quot;       &quot;meta.data&quot;    &quot;active.assay&quot; &quot;active.ident&quot; &quot;graphs&quot;      
##  [6] &quot;neighbors&quot;    &quot;reductions&quot;   &quot;images&quot;       &quot;project.name&quot; &quot;misc&quot;        
## [11] &quot;version&quot;      &quot;commands&quot;     &quot;tools&quot;</code></pre>
<p>Each compartment can be used to store:</p>
<ul>
<li>data from multiple modalities, such as RNAseq (slot <code>assays</code>, sub-slot <code>RNA</code>), ATAC-seq…</li>
<li>results of analyses: dimension reduction, clustering …</li>
<li>etc.</li>
</ul>
<p>You navigate through this hierarchy using <code>@</code> and <code>$</code> signs.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="basic-vizualization.html#cb16-1"></a><span class="kw">slotNames</span>(dataset<span class="op">@</span>assays<span class="op">$</span>RNA)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;        &quot;data&quot;          &quot;scale.data&quot;    &quot;key&quot;          
## [5] &quot;assay.orig&quot;    &quot;var.features&quot;  &quot;meta.features&quot; &quot;misc&quot;</code></pre>
<p>If you want to have a closer look at the data structure of the Seurat object, the R command <code>str()</code> might be very helpful:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="basic-vizualization.html#cb18-1"></a><span class="kw">str</span>(dataset)</span></code></pre></div>
<pre><code>## Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
##   ..@ assays      :List of 1
##   .. ..$ RNA:Formal class &#39;Assay&#39; [package &quot;Seurat&quot;] with 8 slots
##   .. .. .. ..@ counts       :Formal class &#39;dgCMatrix&#39; [package &quot;Matrix&quot;] with 6 slots
##   .. .. .. .. .. ..@ i       : int [1:1194477] 0 1 2 3 4 7 8 11 15 16 ...
##   .. .. .. .. .. ..@ p       : int [1:1298] 0 1445 2686 3841 4946 6084 7162 8255 9308 10342 ...
##   .. .. .. .. .. ..@ Dim     : int [1:2] 1963 1297
##   .. .. .. .. .. ..@ Dimnames:List of 2
##   .. .. .. .. .. .. ..$ : chr [1:1963] &quot;128up&quot; &quot;140up&quot; &quot;14-3-3epsilon&quot; &quot;14-3-3zeta&quot; ...
##   .. .. .. .. .. .. ..$ : chr [1:1297] &quot;GTACTAATTACN_2&quot; &quot;AGGCTAATGGAC_3&quot; &quot;TAGACAAAAGCT_4&quot; &quot;TCTACTAGTGTN_5&quot; ...
##   .. .. .. .. .. ..@ x       : num [1:1194477] 2.57 1.16 8.84 5.94 3.44 ...
##   .. .. .. .. .. ..@ factors : list()
##   .. .. .. ..@ data         :Formal class &#39;dgCMatrix&#39; [package &quot;Matrix&quot;] with 6 slots
##   .. .. .. .. .. ..@ i       : int [1:1194477] 0 1 2 3 4 7 8 11 15 16 ...
##   .. .. .. .. .. ..@ p       : int [1:1298] 0 1445 2686 3841 4946 6084 7162 8255 9308 10342 ...
##   .. .. .. .. .. ..@ Dim     : int [1:2] 1963 1297
##   .. .. .. .. .. ..@ Dimnames:List of 2
##   .. .. .. .. .. .. ..$ : chr [1:1963] &quot;128up&quot; &quot;140up&quot; &quot;14-3-3epsilon&quot; &quot;14-3-3zeta&quot; ...
##   .. .. .. .. .. .. ..$ : chr [1:1297] &quot;GTACTAATTACN_2&quot; &quot;AGGCTAATGGAC_3&quot; &quot;TAGACAAAAGCT_4&quot; &quot;TCTACTAGTGTN_5&quot; ...
##   .. .. .. .. .. ..@ x       : num [1:1194477] 2.57 1.16 8.84 5.94 3.44 ...
##   .. .. .. .. .. ..@ factors : list()
##   .. .. .. ..@ scale.data   : num[0 , 0 ] 
##   .. .. .. ..@ key          : chr &quot;rna_&quot;
##   .. .. .. ..@ assay.orig   : NULL
##   .. .. .. ..@ var.features : logi(0) 
##   .. .. .. ..@ meta.features:&#39;data.frame&#39;:   1963 obs. of  0 variables
##   .. .. .. ..@ misc         : NULL
##   ..@ meta.data   :&#39;data.frame&#39;: 1297 obs. of  3 variables:
##   .. ..$ orig.ident  : Factor w/ 1 level &quot;SeuratProject&quot;: 1 1 1 1 1 1 1 1 1 1 ...
##   .. ..$ nCount_RNA  : num [1:1297] 5057 4711 4728 4562 4569 ...
##   .. ..$ nFeature_RNA: int [1:1297] 1445 1241 1155 1105 1138 1078 1093 1053 1034 1047 ...
##   ..@ active.assay: chr &quot;RNA&quot;
##   ..@ active.ident: Factor w/ 1 level &quot;SeuratProject&quot;: 1 1 1 1 1 1 1 1 1 1 ...
##   .. ..- attr(*, &quot;names&quot;)= chr [1:1297] &quot;GTACTAATTACN_2&quot; &quot;AGGCTAATGGAC_3&quot; &quot;TAGACAAAAGCT_4&quot; &quot;TCTACTAGTGTN_5&quot; ...
##   ..@ graphs      : list()
##   ..@ neighbors   : list()
##   ..@ reductions  : list()
##   ..@ images      : list()
##   ..@ project.name: chr &quot;SeuratProject&quot;
##   ..@ misc        : list()
##   ..@ version     :Classes &#39;package_version&#39;, &#39;numeric_version&#39;  hidden list of 1
##   .. ..$ : int [1:3] 3 2 3
##   ..@ commands    : list()
##   ..@ tools       : list()</code></pre>
<p>The structure of the Seurat object also reflects the pre-processing steps. The scRNAseq workflow consists of 3 main steps :</p>
<ul>
<li>start with raw expression matrix (UMIs after sequencing) and filter out “low quality cells” : data are stored in the compartement <code>counts</code>;</li>
<li>log-normalize the data: to correct for sequencing biais : <code>data</code></li>
<li>perform a dimension reduction such as PCA to reduce the computation time and unnecessary information. PCA requires the data to be normalized and scaled : <code>scale.data</code>.</li>
</ul>
</div>
<div id="perform-dimension-reductions" class="section level2">
<h2><span class="header-section-number">3.3</span> Perform dimension reductions</h2>
<p>When analysing scRNAseq data, you will usually perform some clustering to identify sub-populations and use some technique for visualization (e.g. UMAP).
These analyses are performed on a lower dimensional space.
Here we perform a PCA on the list of highly variable genes provided by the authors.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="basic-vizualization.html#cb20-1"></a>list_hvg  =<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/data_day2/drosophila_scRNAseq/high_var_genes.txt&quot;</span>, <span class="dt">header=</span><span class="ot">FALSE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="basic-vizualization.html#cb21-1"></a>dataset &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(dataset)</span>
<span id="cb21-2"><a href="basic-vizualization.html#cb21-2"></a>dataset &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(dataset, <span class="dt">features =</span> list_hvg<span class="op">$</span>V1)</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="basic-vizualization.html#cb22-1"></a><span class="kw">DimPlot</span>(dataset, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>)</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-pca-1.png" width="672" /></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="basic-vizualization.html#cb23-1"></a><span class="kw">sum</span>(dataset<span class="op">@</span>reductions<span class="op">$</span>pca<span class="op">@</span>stdev[<span class="dv">1</span><span class="op">:</span><span class="dv">35</span>])</span></code></pre></div>
<pre><code>## [1] 49.57188</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="basic-vizualization.html#cb25-1"></a>nPC          =<span class="st"> </span><span class="dv">35</span> <span class="co"># number of PC kept for the analysis</span></span>
<span id="cb25-2"><a href="basic-vizualization.html#cb25-2"></a>n.neighbours =<span class="st"> </span><span class="dv">15</span></span>
<span id="cb25-3"><a href="basic-vizualization.html#cb25-3"></a>resol        =<span class="st"> </span><span class="fl">0.4</span> <span class="co"># impact the number of clusters </span></span>
<span id="cb25-4"><a href="basic-vizualization.html#cb25-4"></a><span class="co"># k-nn graph</span></span>
<span id="cb25-5"><a href="basic-vizualization.html#cb25-5"></a>dataset &lt;-<span class="st"> </span><span class="kw">FindNeighbors</span>(dataset, <span class="dt">k.param =</span> n.neighbours,<span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span>nPC) </span></code></pre></div>
<pre><code>## Computing nearest neighbor graph</code></pre>
<pre><code>## Computing SNN</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="basic-vizualization.html#cb28-1"></a><span class="co"># make the clusters</span></span>
<span id="cb28-2"><a href="basic-vizualization.html#cb28-2"></a>dataset &lt;-<span class="st"> </span><span class="kw">FindClusters</span>(dataset, <span class="dt">resolution =</span> resol) </span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 1297
## Number of edges: 52276
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8298
## Number of communities: 6
## Elapsed time: 0 seconds</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="basic-vizualization.html#cb30-1"></a>dataset &lt;-<span class="st"> </span><span class="kw">RunUMAP</span>(dataset, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span>nPC)</span></code></pre></div>
<pre><code>## Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
## To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
## This message will be shown once per session</code></pre>
<pre><code>## 13:24:14 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
<pre><code>## 13:24:14 Read 1297 rows and found 35 numeric columns</code></pre>
<pre><code>## 13:24:14 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
<pre><code>## 13:24:14 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>## [----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>## **************************************************|
## 13:24:14 Writing NN index file to temp file /tmp/Rtmp0xxBlD/file1f2e4464a8be8c
## 13:24:14 Searching Annoy index using 1 thread, search_k = 3000
## 13:24:15 Annoy recall = 100%
## 13:24:15 Commencing smooth kNN distance calibration using 1 thread
## 13:24:16 Initializing from normalized Laplacian + noise
## 13:24:16 Commencing optimization for 500 epochs, with 50690 positive edges
## 13:24:20 Optimization finished</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="basic-vizualization.html#cb39-1"></a><span class="kw">DimPlot</span>(dataset, <span class="dt">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-umap-1.png" width="672" /></p>
<p>With the function <code>FeaturePlot</code> you can visualize the expression of some gene of interest on th UMAP:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="basic-vizualization.html#cb40-1"></a><span class="kw">FeaturePlot</span>(<span class="dt">object =</span> dataset,</span>
<span id="cb40-2"><a href="basic-vizualization.html#cb40-2"></a>            <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&#39;twi&#39;</span>,<span class="st">&#39;zen&#39;</span>,<span class="st">&#39;aay&#39;</span>, <span class="st">&#39;Antp&#39;</span>),</span>
<span id="cb40-3"><a href="basic-vizualization.html#cb40-3"></a>            <span class="dt">cols =</span> <span class="kw">c</span>(<span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb40-4"><a href="basic-vizualization.html#cb40-4"></a>            <span class="dt">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb40-5"><a href="basic-vizualization.html#cb40-5"></a>            <span class="dt">pt.size =</span> <span class="fl">0.2</span>)</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-genes-1.png" width="672" /></p>
<p>You can define here two genes that you wish to see of the next plots (so you don’t have to change the gene names in the plot functions).</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="basic-vizualization.html#cb41-1"></a>gene1 &lt;-<span class="st"> &#39;twi&#39;</span></span>
<span id="cb41-2"><a href="basic-vizualization.html#cb41-2"></a>gene2 &lt;-<span class="st"> &#39;aay&#39;</span></span></code></pre></div>
<p>Let’s have a look at the correlation between these two genes:</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="basic-vizualization.html#cb42-1"></a><span class="kw">FeatureScatter</span>(<span class="dt">object =</span> dataset, <span class="dt">feature1 =</span> gene1, <span class="dt">feature2 =</span> gene2)</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-co-expression-1.png" width="672" /></p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="basic-vizualization.html#cb43-1"></a>p &lt;-<span class="st"> </span><span class="kw">FeaturePlot</span>(<span class="dt">object =</span> dataset,</span>
<span id="cb43-2"><a href="basic-vizualization.html#cb43-2"></a>         <span class="dt">features =</span> <span class="kw">c</span>(gene1, gene2),</span>
<span id="cb43-3"><a href="basic-vizualization.html#cb43-3"></a>         <span class="dt">cols =</span> <span class="kw">c</span>(<span class="st">&#39;grey&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;blue&#39;</span>),</span>
<span id="cb43-4"><a href="basic-vizualization.html#cb43-4"></a>         <span class="dt">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb43-5"><a href="basic-vizualization.html#cb43-5"></a>         <span class="dt">order =</span> <span class="ot">TRUE</span>,</span>
<span id="cb43-6"><a href="basic-vizualization.html#cb43-6"></a>         <span class="dt">pt.size =</span> <span class="fl">0.5</span>,</span>
<span id="cb43-7"><a href="basic-vizualization.html#cb43-7"></a>         <span class="dt">combine =</span> <span class="ot">FALSE</span>,</span>
<span id="cb43-8"><a href="basic-vizualization.html#cb43-8"></a>         <span class="dt">blend =</span> <span class="ot">TRUE</span>)</span>
<span id="cb43-9"><a href="basic-vizualization.html#cb43-9"></a><span class="kw">print</span>(p[[<span class="dv">3</span>]] <span class="op">+</span><span class="st"> </span><span class="kw">NoLegend</span>())</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-genes-double-1.png" width="672" /></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="basic-vizualization.html#cb44-1"></a>pp &lt;-<span class="st"> </span><span class="kw">FeaturePlot</span>(<span class="dt">object =</span> dataset,</span>
<span id="cb44-2"><a href="basic-vizualization.html#cb44-2"></a>         <span class="dt">features =</span> <span class="kw">c</span>(gene1, gene2),</span>
<span id="cb44-3"><a href="basic-vizualization.html#cb44-3"></a>         <span class="dt">cols =</span> <span class="kw">c</span>(<span class="st">&#39;white&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;blue&#39;</span>),</span>
<span id="cb44-4"><a href="basic-vizualization.html#cb44-4"></a>         <span class="dt">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb44-5"><a href="basic-vizualization.html#cb44-5"></a>         <span class="dt">order =</span> <span class="ot">TRUE</span>,</span>
<span id="cb44-6"><a href="basic-vizualization.html#cb44-6"></a>         <span class="dt">pt.size =</span> <span class="fl">0.5</span>,</span>
<span id="cb44-7"><a href="basic-vizualization.html#cb44-7"></a>         <span class="dt">combine =</span> <span class="ot">FALSE</span>,</span>
<span id="cb44-8"><a href="basic-vizualization.html#cb44-8"></a>         <span class="dt">blend =</span> <span class="ot">TRUE</span>)</span>
<span id="cb44-9"><a href="basic-vizualization.html#cb44-9"></a><span class="kw">print</span>(pp[[<span class="dv">3</span>]] <span class="op">+</span><span class="st"> </span><span class="kw">NoLegend</span>())</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-genes-double-2.png" width="672" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="working-in-python-in-rstudio.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="group-discussion.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
