<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 4 Group discussion | Mapping single-cell RNA-sequencing data on tissue using computations</title>
  <meta name="description" content=" 4 Group discussion | Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content=" 4 Group discussion | Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="git@github.com:PSL-winter-school-2021/02-tuesday-workshop-rna.git" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 4 Group discussion | Mapping single-cell RNA-sequencing data on tissue using computations" />
  
  
  

<meta name="author" content="Lorette Noiret, Nathalie Lehmann, Eric van Leen" />


<meta name="date" content="2021-02-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="basic-vizualization.html"/>
<link rel="next" href="session-info.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="data/notebook/css/custom.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html"><i class="fa fa-check"></i>Introdution</a><ul>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#sources-of-inspiration"><i class="fa fa-check"></i>Sources of inspiration</a></li>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-to-install.html"><a href="what-to-install.html"><i class="fa fa-check"></i><b>1</b> What to install</a></li>
<li class="chapter" data-level="2" data-path="working-in-python-in-rstudio.html"><a href="working-in-python-in-rstudio.html"><i class="fa fa-check"></i><b>2</b> Working in Python in RStudio</a></li>
<li class="chapter" data-level="3" data-path="basic-vizualization.html"><a href="basic-vizualization.html"><i class="fa fa-check"></i><b>3</b> Basic vizualization</a><ul>
<li class="chapter" data-level="3.1" data-path="basic-vizualization.html"><a href="basic-vizualization.html#import-data-in-seurat"><i class="fa fa-check"></i><b>3.1</b> Import data in Seurat</a></li>
<li class="chapter" data-level="3.2" data-path="basic-vizualization.html"><a href="basic-vizualization.html#a-quick-reminder-on-seuratobject"><i class="fa fa-check"></i><b>3.2</b> A quick reminder on SeuratObject</a></li>
<li class="chapter" data-level="3.3" data-path="basic-vizualization.html"><a href="basic-vizualization.html#perform-dimension-reductions"><i class="fa fa-check"></i><b>3.3</b> Perform dimension reductions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="group-discussion.html"><a href="group-discussion.html"><i class="fa fa-check"></i><b>4</b> Group discussion</a><ul>
<li class="chapter" data-level="4.1" data-path="group-discussion.html"><a href="group-discussion.html#interpretation-pc-axes"><i class="fa fa-check"></i><b>4.1</b> Interpretation PC axes</a><ul>
<li class="chapter" data-level="4.1.1" data-path="group-discussion.html"><a href="group-discussion.html#link-with-numi-and-nfeature"><i class="fa fa-check"></i><b>4.1.1</b> Link with nUMI and nFeature</a></li>
<li class="chapter" data-level="4.1.2" data-path="group-discussion.html"><a href="group-discussion.html#enrichement"><i class="fa fa-check"></i><b>4.1.2</b> Enrichement</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>5</b> Session info</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Mapping single-cell RNA-sequencing data on tissue using computations</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="group-discussion" class="section level1">
<h1><span class="header-section-number"> 4</span> Group discussion</h1>
<ul>
<li>How could you identify the genes specifically expressed in one cluster ?</li>
<li>Play with the colors of the plots. How much can this impact the resulting UMAP ?</li>
<li>Try to find genes that are co-expressed (or anti-correlated) in this dataset.</li>
</ul>
<div id="interpretation-pc-axes" class="section level2">
<h2><span class="header-section-number">4.1</span> Interpretation PC axes</h2>
<div id="link-with-numi-and-nfeature" class="section level3">
<h3><span class="header-section-number">4.1.1</span> Link with nUMI and nFeature</h3>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="group-discussion.html#cb45-1"></a><span class="co"># Add PCs coordinates as a metadata</span></span>
<span id="cb45-2"><a href="group-discussion.html#cb45-2"></a>dataset &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(dataset, dataset<span class="op">@</span>reductions<span class="op">$</span>pca<span class="op">@</span>cell.embeddings, <span class="kw">colnames</span>(dataset<span class="op">@</span>reductions<span class="op">$</span>pca<span class="op">@</span>cell.embeddings))</span>
<span id="cb45-3"><a href="group-discussion.html#cb45-3"></a><span class="co"># Add UMAP coordinates as a metadata</span></span>
<span id="cb45-4"><a href="group-discussion.html#cb45-4"></a>dataset &lt;-<span class="st"> </span><span class="kw">AddMetaData</span>(dataset, dataset<span class="op">@</span>reductions<span class="op">$</span>umap<span class="op">@</span>cell.embeddings, <span class="kw">colnames</span>(dataset<span class="op">@</span>reductions<span class="op">$</span>umap<span class="op">@</span>cell.embeddings))</span>
<span id="cb45-5"><a href="group-discussion.html#cb45-5"></a><span class="co"># Check you have the PC aggregated to the metadata</span></span>
<span id="cb45-6"><a href="group-discussion.html#cb45-6"></a><span class="kw">head</span>(dataset[[]])[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##                   orig.ident nCount_RNA nFeature_RNA RNA_snn_res.0.4
## GTACTAATTACN_2 SeuratProject   5056.723         1445               2
## AGGCTAATGGAC_3 SeuratProject   4711.185         1241               0
## TAGACAAAAGCT_4 SeuratProject   4727.516         1155               1
## TCTACTAGTGTN_5 SeuratProject   4561.512         1105               2
## CCGTGCTGAACA_3 SeuratProject   4569.284         1138               0
## GAGAAAATGAAG_7 SeuratProject   4528.382         1078               1
##                seurat_clusters       PC_1       PC_2       PC_3     PC_4
## GTACTAATTACN_2               2  0.8901358  1.7567252 -2.6676502 5.054322
## AGGCTAATGGAC_3               0  2.0935612  2.6764714 -1.2593752 1.026615
## TAGACAAAAGCT_4               1 -2.0516447 -0.1691875  2.3202232 4.824665
## TCTACTAGTGTN_5               2  0.8153851  1.6728201 -5.6104631 2.608583
## CCGTGCTGAACA_3               0 -1.0697989  2.3039091  2.9484302 1.376591
## GAGAAAATGAAG_7               1 -1.1697802  0.1742288  0.9408307 1.676755
##                       PC_5
## GTACTAATTACN_2  0.33856502
## AGGCTAATGGAC_3  0.31540995
## TAGACAAAAGCT_4  0.01278697
## TCTACTAGTGTN_5 -0.50347780
## CCGTGCTGAACA_3  0.08532385
## GAGAAAATGAAG_7 -2.77752873</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="group-discussion.html#cb47-1"></a><span class="kw">FeaturePlot</span>(<span class="dt">object =</span> dataset, <span class="dt">features =</span> <span class="st">&quot;nCount_RNA&quot;</span>,<span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>)</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-seqDepth-1.png" width="672" /></p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="group-discussion.html#cb48-1"></a><span class="kw">FeaturePlot</span>(<span class="dt">object =</span> dataset, <span class="dt">features =</span> <span class="st">&quot;nFeature_RNA&quot;</span>,<span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>)</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-seqDepth-2.png" width="672" /></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="group-discussion.html#cb49-1"></a><span class="kw">ggplot</span>(dataset[[]],</span>
<span id="cb49-2"><a href="group-discussion.html#cb49-2"></a>       <span class="kw">aes</span>(<span class="dt">x=</span>PC_<span class="dv">1</span>, <span class="dt">y=</span>PC_<span class="dv">2</span>, <span class="dt">color =</span> seurat_clusters)) <span class="op">+</span></span>
<span id="cb49-3"><a href="group-discussion.html#cb49-3"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">size =</span> nCount_RNA)) <span class="op">+</span></span>
<span id="cb49-4"><a href="group-discussion.html#cb49-4"></a><span class="st">  </span><span class="kw">geom_rug</span>()</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-seqDepth-3.png" width="672" /></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="group-discussion.html#cb50-1"></a><span class="kw">ggplot</span>(dataset[[]],</span>
<span id="cb50-2"><a href="group-discussion.html#cb50-2"></a>       <span class="kw">aes</span>(<span class="dt">x=</span>UMAP_<span class="dv">1</span>, <span class="dt">y=</span>UMAP_<span class="dv">2</span>, <span class="dt">color =</span> seurat_clusters)) <span class="op">+</span></span>
<span id="cb50-3"><a href="group-discussion.html#cb50-3"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">size =</span> nCount_RNA)) <span class="op">+</span></span>
<span id="cb50-4"><a href="group-discussion.html#cb50-4"></a><span class="st">  </span><span class="kw">geom_rug</span>()</span></code></pre></div>
<p><img src="02-tuesday-workshop-rna_files/figure-html/plot-seqDepth-4.png" width="672" /></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="group-discussion.html#cb51-1"></a><span class="co"># Extract PC gene loadings for the 10 first components</span></span>
<span id="cb51-2"><a href="group-discussion.html#cb51-2"></a>PC_df &lt;-<span class="st"> </span>dataset<span class="op">@</span>reductions<span class="op">$</span>pca<span class="op">@</span>feature.loadings[,<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</span></code></pre></div>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="group-discussion.html#cb52-1"></a><span class="co"># Select PC to work with</span></span>
<span id="cb52-2"><a href="group-discussion.html#cb52-2"></a>nPC &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb52-3"><a href="group-discussion.html#cb52-3"></a><span class="co"># Select highest 5 genes in PC1</span></span>
<span id="cb52-4"><a href="group-discussion.html#cb52-4"></a>list_topVarPC1_plus &lt;-<span class="st"> </span><span class="kw">sort</span>(PC_df[,<span class="kw">paste0</span>(<span class="st">&#39;PC_&#39;</span>, nPC)], <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span>
<span id="cb52-5"><a href="group-discussion.html#cb52-5"></a>list_topVarPC1_minus &lt;-<span class="st"> </span><span class="kw">sort</span>(PC_df[,<span class="kw">paste0</span>(<span class="st">&#39;PC_&#39;</span>, nPC)], <span class="dt">decreasing =</span> <span class="ot">FALSE</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span>
<span id="cb52-6"><a href="group-discussion.html#cb52-6"></a>list_topVarPC1_plus</span></code></pre></div>
<pre><code>##       Ptr       sdt       aop    5-HT2A      veil 
## 0.2041575 0.1794579 0.1573053 0.1562678 0.1454172</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="group-discussion.html#cb54-1"></a>list_topVarPC1_minus</span></code></pre></div>
<pre><code>##                           twi ventrally-expressed-protein-D 
##                    -0.2207021                    -0.2158339 
##                           sna                        stumps 
##                    -0.2152931                    -0.2063053 
##                          zfh1 
##                    -0.2051081</code></pre>
</div>
<div id="enrichement" class="section level3">
<h3><span class="header-section-number">4.1.2</span> Enrichement</h3>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="group-discussion.html#cb56-1"></a>CallGroupGo =<span class="st"> </span><span class="cf">function</span>(list_genes){</span>
<span id="cb56-2"><a href="group-discussion.html#cb56-2"></a><span class="co"># groupGO return the GO profile at a specific level·</span></span>
<span id="cb56-3"><a href="group-discussion.html#cb56-3"></a>eg =<span class="st"> </span><span class="kw">bitr</span>(list_genes, <span class="dt">fromType=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="dt">toType=</span><span class="kw">c</span>(<span class="st">&quot;FLYBASE&quot;</span>,<span class="st">&quot;ENTREZID&quot;</span>), <span class="dt">OrgDb=</span><span class="st">&quot;org.Dm.eg.db&quot;</span>)</span>
<span id="cb56-4"><a href="group-discussion.html#cb56-4"></a><span class="co"># copy to clipboard for comparison with online app</span></span>
<span id="cb56-5"><a href="group-discussion.html#cb56-5"></a><span class="co">#write_clip(list_topVarPC1)</span></span>
<span id="cb56-6"><a href="group-discussion.html#cb56-6"></a>ggo &lt;-<span class="st"> </span><span class="kw">groupGO</span>(<span class="dt">gene     =</span> <span class="kw">na.omit</span>(eg<span class="op">$</span>ENTREZID),</span>
<span id="cb56-7"><a href="group-discussion.html#cb56-7"></a>               <span class="dt">OrgDb    =</span> org.Dm.eg.db,</span>
<span id="cb56-8"><a href="group-discussion.html#cb56-8"></a>               <span class="dt">ont      =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb56-9"><a href="group-discussion.html#cb56-9"></a>               <span class="dt">level    =</span> <span class="dv">3</span>,</span>
<span id="cb56-10"><a href="group-discussion.html#cb56-10"></a>               <span class="dt">readable =</span> <span class="ot">TRUE</span>)</span>
<span id="cb56-11"><a href="group-discussion.html#cb56-11"></a><span class="kw">return</span>(ggo)</span>
<span id="cb56-12"><a href="group-discussion.html#cb56-12"></a>}</span>
<span id="cb56-13"><a href="group-discussion.html#cb56-13"></a></span>
<span id="cb56-14"><a href="group-discussion.html#cb56-14"></a>CallEnrichGo =<span class="st"> </span><span class="cf">function</span>(list_genes,pval,qval){</span>
<span id="cb56-15"><a href="group-discussion.html#cb56-15"></a><span class="co">#return the enrichment GO categories after FDR control.·</span></span>
<span id="cb56-16"><a href="group-discussion.html#cb56-16"></a>eg =<span class="st"> </span><span class="kw">bitr</span>(list_genes, <span class="dt">fromType=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="dt">toType=</span><span class="kw">c</span>(<span class="st">&quot;FLYBASE&quot;</span>,<span class="st">&quot;ENTREZID&quot;</span>), <span class="dt">OrgDb=</span><span class="st">&quot;org.Dm.eg.db&quot;</span>)</span>
<span id="cb56-17"><a href="group-discussion.html#cb56-17"></a></span>
<span id="cb56-18"><a href="group-discussion.html#cb56-18"></a>ego &lt;-<span class="st"> </span><span class="kw">enrichGO</span>(<span class="dt">gene          =</span> <span class="kw">na.omit</span>(eg<span class="op">$</span>ENTREZID),</span>
<span id="cb56-19"><a href="group-discussion.html#cb56-19"></a>                <span class="dt">OrgDb         =</span> org.Dm.eg.db,</span>
<span id="cb56-20"><a href="group-discussion.html#cb56-20"></a>                <span class="dt">ont           =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb56-21"><a href="group-discussion.html#cb56-21"></a>                <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb56-22"><a href="group-discussion.html#cb56-22"></a>                <span class="dt">pvalueCutoff  =</span> pval,</span>
<span id="cb56-23"><a href="group-discussion.html#cb56-23"></a>                <span class="dt">qvalueCutoff  =</span> qval,</span>
<span id="cb56-24"><a href="group-discussion.html#cb56-24"></a>        <span class="dt">readable      =</span> <span class="ot">TRUE</span>)</span>
<span id="cb56-25"><a href="group-discussion.html#cb56-25"></a><span class="kw">return</span>(ego)</span>
<span id="cb56-26"><a href="group-discussion.html#cb56-26"></a>}</span>
<span id="cb56-27"><a href="group-discussion.html#cb56-27"></a></span>
<span id="cb56-28"><a href="group-discussion.html#cb56-28"></a>CallKeggPath =<span class="st"> </span><span class="cf">function</span>(list_genes){</span>
<span id="cb56-29"><a href="group-discussion.html#cb56-29"></a><span class="co">#return the enrichment GO categories after FDR control.·</span></span>
<span id="cb56-30"><a href="group-discussion.html#cb56-30"></a>eg =<span class="st"> </span><span class="kw">bitr</span>(list_genes, <span class="dt">fromType=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="dt">toType=</span><span class="kw">c</span>(<span class="st">&quot;FLYBASE&quot;</span>,<span class="st">&quot;ENTREZID&quot;</span>), <span class="dt">OrgDb=</span><span class="st">&quot;org.Dm.eg.db&quot;</span>)</span>
<span id="cb56-31"><a href="group-discussion.html#cb56-31"></a></span>
<span id="cb56-32"><a href="group-discussion.html#cb56-32"></a>kk&lt;-<span class="st"> </span><span class="kw">enrichKEGG</span>(<span class="dt">gene          =</span> <span class="kw">na.omit</span>(eg<span class="op">$</span>ENTREZID),</span>
<span id="cb56-33"><a href="group-discussion.html#cb56-33"></a>                <span class="dt">organism      =</span> org.Dm.eg.db,</span>
<span id="cb56-34"><a href="group-discussion.html#cb56-34"></a>                <span class="dt">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb56-35"><a href="group-discussion.html#cb56-35"></a>                <span class="dt">use_internal_data =</span> <span class="ot">TRUE</span>)</span>
<span id="cb56-36"><a href="group-discussion.html#cb56-36"></a><span class="kw">return</span>(kk)</span>
<span id="cb56-37"><a href="group-discussion.html#cb56-37"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="group-discussion.html#cb57-1"></a>ego1p =<span class="st"> </span><span class="kw">CallEnrichGo</span>(list_topVarPC1_plus,<span class="fl">0.01</span>,<span class="fl">0.05</span>)</span>
<span id="cb57-2"><a href="group-discussion.html#cb57-2"></a>ego1m =<span class="st"> </span><span class="kw">CallEnrichGo</span>(list_topVarPC1_minus,<span class="fl">0.01</span>,<span class="fl">0.05</span>)</span>
<span id="cb57-3"><a href="group-discussion.html#cb57-3"></a></span>
<span id="cb57-4"><a href="group-discussion.html#cb57-4"></a>ego2 =<span class="st"> </span><span class="kw">CallEnrichGo</span>(list_topVarPC2,<span class="fl">0.02</span>,<span class="fl">0.15</span>)</span>
<span id="cb57-5"><a href="group-discussion.html#cb57-5"></a>ego2p =<span class="st"> </span><span class="kw">CallEnrichGo</span>(list_topVarPC2_plus,<span class="fl">0.01</span>,<span class="fl">0.05</span>)</span>
<span id="cb57-6"><a href="group-discussion.html#cb57-6"></a>ego2m =<span class="st"> </span><span class="kw">CallEnrichGo</span>(list_topVarPC2_minus,<span class="fl">0.01</span>,<span class="fl">0.05</span>)</span>
<span id="cb57-7"><a href="group-discussion.html#cb57-7"></a></span>
<span id="cb57-8"><a href="group-discussion.html#cb57-8"></a>ego3 =<span class="st"> </span><span class="kw">CallEnrichGo</span>(list_topVarPC3,<span class="fl">0.02</span>,<span class="fl">0.15</span>)</span>
<span id="cb57-9"><a href="group-discussion.html#cb57-9"></a>ego3p =<span class="st"> </span><span class="kw">CallEnrichGo</span>(list_topVarPC3_plus,<span class="fl">0.01</span>,<span class="fl">0.05</span>)</span>
<span id="cb57-10"><a href="group-discussion.html#cb57-10"></a>ego3m =<span class="st"> </span><span class="kw">CallEnrichGo</span>(list_topVarPC3_minus,<span class="fl">0.01</span>,<span class="fl">0.05</span>)</span></code></pre></div>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="group-discussion.html#cb58-1"></a><span class="co">#barplot(ggo1, drop=TRUE, showCategory=10,main = &quot;Genes associated with PC1&quot;)</span></span>
<span id="cb58-2"><a href="group-discussion.html#cb58-2"></a><span class="kw">barplot</span>(ego1, <span class="dt">showCategory=</span><span class="dv">25</span>, <span class="dt">main =</span><span class="st">&quot;enrich genes PC1&quot;</span>)</span>
<span id="cb58-3"><a href="group-discussion.html#cb58-3"></a><span class="kw">barplot</span>(ego1p, <span class="dt">showCategory=</span><span class="dv">25</span>, <span class="dt">main =</span><span class="st">&quot;enrich genes PC1&quot;</span>)</span>
<span id="cb58-4"><a href="group-discussion.html#cb58-4"></a><span class="kw">barplot</span>(ego1m, <span class="dt">showCategory=</span><span class="dv">25</span>, <span class="dt">main =</span><span class="st">&quot;enrich genes PC1&quot;</span>)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="basic-vizualization.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="session-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
