<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Mapping single-cell RNA-sequencing data on tissue using computations</title>
  <meta name="description" content="Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="git@github.com:PSL-winter-school-2021/02-tuesday-workshop-rna.git" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Mapping single-cell RNA-sequencing data on tissue using computations" />
  
  
  

<meta name="author" content="Lorette Noiret, Nathalie Lehmann, Eric van Leen" />


<meta name="date" content="2021-02-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="what-to-install.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="data/notebook/css/custom.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html"><i class="fa fa-check"></i>Introdution</a>
<ul>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#sources-of-inspiration"><i class="fa fa-check"></i>Sources of inspiration</a></li>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-to-install.html"><a href="what-to-install.html"><i class="fa fa-check"></i><b>1</b> What to install</a></li>
<li class="chapter" data-level="2" data-path="working-in-python-in-rstudio.html"><a href="working-in-python-in-rstudio.html"><i class="fa fa-check"></i><b>2</b> Working in Python in RStudio</a></li>
<li class="chapter" data-level="3" data-path="basic-vizualization.html"><a href="basic-vizualization.html"><i class="fa fa-check"></i><b>3</b> Basic vizualization</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-vizualization.html"><a href="basic-vizualization.html#import-data-in-seurat"><i class="fa fa-check"></i><b>3.1</b> Import data in Seurat</a></li>
<li class="chapter" data-level="3.2" data-path="basic-vizualization.html"><a href="basic-vizualization.html#a-quick-reminder-on-seuratobject"><i class="fa fa-check"></i><b>3.2</b> A quick reminder on SeuratObject</a></li>
<li class="chapter" data-level="3.3" data-path="basic-vizualization.html"><a href="basic-vizualization.html#perform-dimension-reductions"><i class="fa fa-check"></i><b>3.3</b> Perform dimension reductions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html"><i class="fa fa-check"></i><b>4</b> Spatial transcriptome with novoSpaRc</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#import-data"><i class="fa fa-check"></i><b>4.1</b> Import data</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#single-cells-expression-matrix"><i class="fa fa-check"></i><b>4.1.1</b> Single cells expression matrix</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#create-the-geometry-of-your-tissue-target-space"><i class="fa fa-check"></i><b>4.2</b> Create the geometry of your tissue (<em>target space</em>)</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#case-1-you-know-the-cellular-locations"><i class="fa fa-check"></i><b>4.2.1</b> case 1 : you know the cellular locations</a></li>
<li class="chapter" data-level="4.2.2" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#alternative-tissue-spaces"><i class="fa fa-check"></i><b>4.2.2</b> Alternative tissue spaces</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="predict-the-spatial-transcriptome.html"><a href="predict-the-spatial-transcriptome.html"><i class="fa fa-check"></i><b>5</b> 3. Predict the spatial transcriptome</a>
<ul>
<li class="chapter" data-level="5.1" data-path="predict-the-spatial-transcriptome.html"><a href="predict-the-spatial-transcriptome.html#predict-the-spatial-transcriptome-without-landmark-genes"><i class="fa fa-check"></i><b>5.1</b> Predict the spatial transcriptome (without landmark genes)</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>6</b> Session info</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Mapping single-cell RNA-sequencing data on tissue using computations</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Mapping single-cell RNA-sequencing data on tissue using computations</h1>
<p class="author"><em>Lorette Noiret, Nathalie Lehmann, Eric van Leen</em></p>
<p class="date"><em>2021-02-08</em></p>
</div>
<div id="introdution" class="section level1 unnumbered">
<h1>Introdution</h1>
<p><strong>Objectives</strong> :</p>
<p>Yesterday, we saw how to pre-process single cell rna sequencing data (Drop-seq) to extract “high quality cells.” Today, we will explore further a similar dataset and use a computional approach called <em>novoSparc</em> to reconstruct the spatial transcriptome. Before performing an advanced technic such as spatial transcriptomics, it is good to have some kind of validation. So as the first step, we will visualize co-expressed genes in UMAP and see if we can find the same co-expression in our predicted spatial transcriptome.</p>
<p>In this pratical session, you will learn :</p>
<ul>
<li>how to visualize co-expressed genes in UMAP ;</li>
<li>how to map dissociated single cells back to their tissue of origin using novoSpaRc (de novo Spatial Reconstruction of Single-Cell Gene Expression).</li>
</ul>
<p>We will be using one of the dataset publilshied with Nitizan et al (Nature 2019). The data are alreay filtered (“high quality cells”) and log-normalized.</p>
<div id="sources-of-inspiration" class="section level2 unnumbered">
<h2>Sources of inspiration</h2>
<p>This course material is based on Novosparc tutorial on github:</p>
<ul>
<li><a href="https://github.com/rajewsky-lab/novosparc">novosparc tutorial</a>: reconstruct_drosophila_embryo_tutorial.ipynb</li>
</ul>
</div>
<div id="bibliography" class="section level2 unnumbered">
<h2>Bibliography</h2>
<ul>
<li>Karaiskos et al.The Drosophila embryo at single-cell transcriptome resolution. Science13 Oct 2017 : 194-199</li>
<li>Nitzan et al. Gene expression cartography. Nature.2019 Dec;576(7785):132-137.<br />
</li>
<li>Tutorial <a href="https://bookdown.org/yihui/rmarkdown-cookbook/eng-python.html">working with Python in R</a></li>
</ul>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="what-to-install.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
