<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 4 3. Predict the spatial transcriptome | Mapping single-cell RNA-sequencing data on tissue using computations</title>
  <meta name="description" content=" 4 3. Predict the spatial transcriptome | Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content=" 4 3. Predict the spatial transcriptome | Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="git@github.com:PSL-winter-school-2021/02-tuesday-workshop-rna.git" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 4 3. Predict the spatial transcriptome | Mapping single-cell RNA-sequencing data on tissue using computations" />
  
  
  

<meta name="author" content="Lorette Noiret, Nathalie Lehmann, Eric van Leen" />


<meta name="date" content="2021-02-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="spatial-transcriptome-with-novosparc.html"/>
<link rel="next" href="session-info.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="data/notebook/css/custom.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html"><i class="fa fa-check"></i>Introdution</a>
<ul>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#sources-of-inspiration"><i class="fa fa-check"></i>Sources of inspiration</a></li>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="working-in-python-in-rstudio.html"><a href="working-in-python-in-rstudio.html"><i class="fa fa-check"></i><b>1</b> Working in Python in RStudio</a></li>
<li class="chapter" data-level="2" data-path="basic-vizualization.html"><a href="basic-vizualization.html"><i class="fa fa-check"></i><b>2</b> Basic vizualization</a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-vizualization.html"><a href="basic-vizualization.html#import-data-in-seurat"><i class="fa fa-check"></i><b>2.1</b> Import data in Seurat</a></li>
<li class="chapter" data-level="2.2" data-path="basic-vizualization.html"><a href="basic-vizualization.html#a-quick-reminder-on-seuratobject"><i class="fa fa-check"></i><b>2.2</b> A quick reminder on SeuratObject</a></li>
<li class="chapter" data-level="2.3" data-path="basic-vizualization.html"><a href="basic-vizualization.html#perform-dimension-reductions"><i class="fa fa-check"></i><b>2.3</b> Perform dimension reductions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html"><i class="fa fa-check"></i><b>3</b> Spatial transcriptome with novoSparc</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#import-data"><i class="fa fa-check"></i><b>3.1</b> Import data</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#single-cells-expression-matrix"><i class="fa fa-check"></i><b>3.1.1</b> Single cells expression matrix</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#create-the-geometry-of-your-tissue-target-space"><i class="fa fa-check"></i><b>3.2</b> Create the geometry of your tissue (<em>target space</em>)</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#case-1-you-know-the-cellular-locations"><i class="fa fa-check"></i><b>3.2.1</b> case 1 : you know the cellular locations</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#alternative-tissue-spaces"><i class="fa fa-check"></i><b>3.2.2</b> Alternative tissue spaces</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="predict-the-spatial-transcriptome.html"><a href="predict-the-spatial-transcriptome.html"><i class="fa fa-check"></i><b>4</b> 3. Predict the spatial transcriptome</a>
<ul>
<li class="chapter" data-level="4.1" data-path="predict-the-spatial-transcriptome.html"><a href="predict-the-spatial-transcriptome.html#predict-the-spatial-transcriptome-without-landmark-genes"><i class="fa fa-check"></i><b>4.1</b> Predict the spatial transcriptome (without landmark genes)</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>5</b> Session info</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Mapping single-cell RNA-sequencing data on tissue using computations</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="predict-the-spatial-transcriptome" class="section level1" number="4">
<h1><span class="header-section-number"> 4</span> 3. Predict the spatial transcriptome</h1>
<div id="predict-the-spatial-transcriptome-without-landmark-genes" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Predict the spatial transcriptome (without landmark genes)</h2>
<p>Once you have defined the cellular locations, you can create a “tissue” object that will contain the predicted spatial transcriptome.</p>
<pre class="python3"><code># create a tissue object, the results will be saved in the directory &quot;outputs&quot;
tissue = novosparc.cm.Tissue(dataset=dataset, locations=locations, output_folder=&#39;outputs&#39;) 
</code></pre>
<p><strong>Principle of novosparc reconstruction</strong> (if no landmarks genes are provided) :</p>
<ol style="list-style-type: decimal">
<li><p>Construct a k-nearest neighbors (knn) graph</p>
<ul>
<li>for the physical space (cellular location) : calculate the (euclidian) distance matrix between the locations, then for each location only keep the k-nearest neighbours (the value k is set by <em>num_neighbors_s</em>).</li>
<li>for the expression space (single cells) : calculate the distance (correlation) matrix between the each cells, then for each location only keep the k-nearest neighbours (the value k is set by <em>num_neighbors_t</em>)</li>
</ul></li>
<li><p>Computes the shortest path lengths for each pair of cells, this step results in graph-based distance matrices for cells and for locations.</p></li>
</ol>
<ol start="3" style="list-style-type: decimal">
<li>Match the N single cells to the M positions using these graphs: the algorithm try to minimize the discrepancy between the pair-wise graph-based distances in expression space and in physical space</li>
</ol>
<pre class="python3"><code>tissue.setup_reconstruction(num_neighbors_s = 5, num_neighbors_t = 5)   # compute the knn graphs                         </code></pre>
<p>Now we can reconstruct our tissue. The alpha_linear parameter controls the contribution of the two branches in novoSpaRc: alpha_linear=0 means that no prior information is available</p>
<pre class="python3"><code>tissue.reconstruct(alpha_linear=0, epsilon=5e-3)
</code></pre>
<p>We store the spatial gene expression matrix (sDGE) for the highly variable genes across all locations in a new variable</p>
<pre class="python3"><code>sdge_hvg = tissue.sdge</code></pre>
<p>Plot the predicted patterns</p>
<pre class="python3"><code>plt.figure(figsize=(12,4))
plt.subplot(231)
plt.scatter(locations[:, 0], locations[:, 1], c=sdge_hvg[np.argwhere(hvg == &#39;ftz&#39;)[0], :].flatten(), s=1)
plt.axis(&#39;off&#39;)
plt.subplot(232)
plt.scatter(locations[:, 0], locations[:, 1], c=sdge_hvg[np.argwhere(hvg == &#39;twi&#39;)[0], :].flatten(), s=1)
plt.axis(&#39;off&#39;)
plt.subplot(233)
plt.scatter(locations[:, 0], locations[:, 1], c=sdge_hvg[np.argwhere(hvg == &#39;zen&#39;)[0], :].flatten(), s=1)
plt.axis(&#39;off&#39;)
plt.subplot(234)
plt.scatter(locations[:, 0], locations[:, 1], c=sdge_hvg[np.argwhere(hvg == &#39;ImpE2&#39;)[0], :].flatten(), s=1)
plt.axis(&#39;off&#39;)
plt.subplot(235)
plt.scatter(locations[:, 0], locations[:, 1], c=sdge_hvg[np.argwhere(hvg == &#39;sim&#39;)[0], :].flatten(), s=1)
plt.axis(&#39;off&#39;)
plt.subplot(236)
plt.scatter(locations[:, 0], locations[:, 1], c=sdge_hvg[np.argwhere(hvg == &#39;pyr&#39;)[0], :].flatten(), s=1)
plt.axis(&#39;off&#39;)
plt.show()
plt.close()</code></pre>
<p>This reconstruction has been realised without priori information. We can greatly improve the prediction by adding landmark genes
## Predict the spatial transcriptome (with landmark genes)
### Import landmark genes</p>
<pre class="python3"><code>insitu_genes = np.genfromtxt(datafile_atlas, usecols=range(84), dtype=&#39;str&#39;, max_rows=1)
atlas_matrix = np.loadtxt(datafile_atlas, usecols=range(84), skiprows=1)
</code></pre>
<pre class="python3"><code># je prends seulement 3 genes
# faudrait filter les colonnes de atlas_matrix pour ietre sure que les noms
# de genes sont les memes que ceux dans single cells
insitu_genes = insitu_genes[2:5]
atlas_matrix = atlas_matrix[:,2:5]</code></pre>
<p>On cherche les colonnes dans les données single cells qui matchent</p>
<pre class="python3"><code>gene_names = tissue_with_markers.gene_names
markers_in_sc = np.array([], dtype=&#39;int&#39;) # indice marker dans table sc

for marker in insitu_genes:
    marker_index = np.where(gene_names == marker)[0]
    if len(marker_index) &gt; 0:
        markers_in_sc = np.append(markers_in_sc, marker_index[0])

print(markers_in_sc.shape )
print(gene_names[markers_in_sc])</code></pre>
<p>We are going to create a new tissue object and specify the markers to use</p>
<pre class="python3"><code># contrairement au tutoriel, je ne pense pas qu&#39;on doivent specifier la
# amtrice de marker quand on crée l&#39;ebjet tissue
tissue_with_markers = novosparc.cm.Tissue(dataset=dataset, locations=locations)

tissue_with_markers.setup_reconstruction(markers_to_use=markers_in_sc, insitu_matrix=atlas_matrix,num_neighbors_s = 5, num_neighbors_t = 5)   # compute the knn graphs                         
</code></pre>
<p>alpha_linear = 0.5</p>
<pre class="python3"><code>alpha_linear = 0.5
tissue_with_markers.reconstruct(alpha_linear=0.5, epsilon=5e-3)</code></pre>
<pre class="python3"><code>sdge_hvg2 = tissue_with_markers.sdge </code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="spatial-transcriptome-with-novosparc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="session-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
