<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 3 Spatial transcriptome with novoSparc | Mapping single-cell RNA-sequencing data on tissue using computations</title>
  <meta name="description" content=" 3 Spatial transcriptome with novoSparc | Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content=" 3 Spatial transcriptome with novoSparc | Mapping single-cell RNA-sequencing data on tissue using computations" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="git@github.com:PSL-winter-school-2021/02-tuesday-workshop-rna.git" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 3 Spatial transcriptome with novoSparc | Mapping single-cell RNA-sequencing data on tissue using computations" />
  
  
  

<meta name="author" content="Lorette Noiret, Nathalie Lehmann, Eric van Leen" />


<meta name="date" content="2021-02-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="basic-vizualization.html"/>
<link rel="next" href="predict-the-spatial-transcriptome.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="data/notebook/css/custom.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html"><i class="fa fa-check"></i>Introdution</a><ul>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#sources-of-inspiration"><i class="fa fa-check"></i>Sources of inspiration</a></li>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="working-in-python-in-rstudio.html"><a href="working-in-python-in-rstudio.html"><i class="fa fa-check"></i><b>1</b> Working in Python in RStudio</a></li>
<li class="chapter" data-level="2" data-path="basic-vizualization.html"><a href="basic-vizualization.html"><i class="fa fa-check"></i><b>2</b> Basic vizualization</a><ul>
<li class="chapter" data-level="2.1" data-path="basic-vizualization.html"><a href="basic-vizualization.html#import-data-in-seurat"><i class="fa fa-check"></i><b>2.1</b> Import data in Seurat</a></li>
<li class="chapter" data-level="2.2" data-path="basic-vizualization.html"><a href="basic-vizualization.html#a-quick-reminder-on-seuratobject"><i class="fa fa-check"></i><b>2.2</b> A quick reminder on SeuratObject</a></li>
<li class="chapter" data-level="2.3" data-path="basic-vizualization.html"><a href="basic-vizualization.html#perform-dimension-reductions"><i class="fa fa-check"></i><b>2.3</b> Perform dimension reductions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html"><i class="fa fa-check"></i><b>3</b> Spatial transcriptome with novoSparc</a><ul>
<li class="chapter" data-level="3.1" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#import-data"><i class="fa fa-check"></i><b>3.1</b> Import data</a><ul>
<li class="chapter" data-level="3.1.1" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#single-cells-expression-matrix"><i class="fa fa-check"></i><b>3.1.1</b> Single cells expression matrix</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#create-the-geometry-of-your-tissue-target-space"><i class="fa fa-check"></i><b>3.2</b> Create the geometry of your tissue (<em>target space</em>)</a><ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#case-1-you-know-the-cellular-locations"><i class="fa fa-check"></i><b>3.2.1</b> case 1 : you know the cellular locations</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-transcriptome-with-novosparc.html"><a href="spatial-transcriptome-with-novosparc.html#alternative-tissue-spaces"><i class="fa fa-check"></i><b>3.2.2</b> Alternative tissue spaces</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="predict-the-spatial-transcriptome.html"><a href="predict-the-spatial-transcriptome.html"><i class="fa fa-check"></i><b>4</b> 3. Predict the spatial transcriptome</a><ul>
<li class="chapter" data-level="4.1" data-path="predict-the-spatial-transcriptome.html"><a href="predict-the-spatial-transcriptome.html#predict-the-spatial-transcriptome-without-landmark-genes"><i class="fa fa-check"></i><b>4.1</b> Predict the spatial transcriptome (without landmark genes)</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>5</b> Session info</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Mapping single-cell RNA-sequencing data on tissue using computations</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="spatial-transcriptome-with-novosparc" class="section level1">
<h1><span class="header-section-number"> 3</span> Spatial transcriptome with novoSparc</h1>
<div id="import-data" class="section level2">
<h2><span class="header-section-number">3.1</span> Import data</h2>
<p>novoSpaRc is an algorithm that predicts the location of dissociated single cells. To work, the algorithm needs :</p>
<ul>
<li>the expression matrix of the single cells data</li>
<li>the shape/geometry of the tissue of origin</li>
</ul>
<p>The performance is greatly increased if you provide some marker genes (also called landmarks / atlas) for which the expression patterns are known.</p>
<div id="single-cells-expression-matrix" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Single cells expression matrix</h3>
<pre class="python3"><code># Single data : expression matrix (already imported)
datafile_sc      =&#39;data/data_day2/drosophila_scRNAseq/dge_normalized.txt&#39;

# Optional : provide a list of highly variable genes (speed up the calculations if we only map a small number of genes)
datafile_sc_hvg  =&#39;data/data_day2/drosophila_scRNAseq/high_var_genes.txt&#39;
# Optional : geometry of the tissue
datafile_geom   =&#39;data/data_day2/bdtnp/geometry.txt&#39;
# Optional : Landamark genes = genes with known pattern of expression (eg. FISH image...)
datafile_atlas  =&#39;data/data_day2/bdtnp/atlas.txt&#39;</code></pre>
<p>How to format the single cell matrix :</p>
<ul>
<li>Novosparc expect a matrix with <strong>genes in columns and cells in rows</strong>. Be careful, this is not the usual order !</li>
<li>{#Nitizan} and al. advised to use the <strong>normalized gene expression matrix</strong> (e.g. using the global scaling log-normalization method)</li>
<li>The data need to be in the <code>Anndata format. Novosparc has a function to import an expression matrix and to convert it in the</code>Anndata<code>format (alternativaly you can convert a pandas dataframe to an</code>Anndata` easily).</li>
</ul>
<pre class="python3"><code>import novosparc
dataset   = novosparc.io.load_data(datafile_sc)

dataset   = dataset.T  # in this example we have to tanspose the matrix to have gene in columns and cells in rows
print (&#39;number of cells: &#39;,dataset.shape[0],&#39;\nnumber of genes: &#39;, dataset.shape[1])
print(&quot;Gene names :&quot;, dataset.var.index.tolist()[:10])

# Alternative : convert a pandas dataframe to AnnData format
# data = pd.read_csv(data_path,sep=&#39;\t&#39;) 
# dataset = sc.AnnData(data )
# dataset = dataset2.T </code></pre>
<p>To save computation time, we will only map a subset of the genes. Here we will focus on a list of highly variable genes provided with the publication.</p>
<pre class="python3"><code>hvg           = np.genfromtxt(datafile_sc_hvg, dtype=&#39;str&#39;)

dataset,hvg   = novosparc.pp.subset_to_hvg(dataset, gene_list = hvg )

dataset.shape</code></pre>
</div>
</div>
<div id="create-the-geometry-of-your-tissue-target-space" class="section level2">
<h2><span class="header-section-number">3.2</span> Create the geometry of your tissue (<em>target space</em>)</h2>
<p>You need to create the grid that describes the shape of your tissue (location of the cells in the tissue).</p>
<p>You have 3 options :</p>
<ul>
<li>either 1) you know the cellular locations and have them in a file ;</li>
<li>either 2) you don’t know the cellular locations, but you know the shape of your tissue;</li>
<li>either 3) you know nothing and will just assume that your tissue is a rectangle.</li>
</ul>
<div id="case-1-you-know-the-cellular-locations" class="section level3">
<h3><span class="header-section-number">3.2.1</span> case 1 : you know the cellular locations</h3>
<p>We suppose that you have a file that contain the XYZ location of each cells in your tissue.</p>
<p>Note : if your tissue is flat, you can set the Z-coordinate to 1.</p>
<pre class="python3"><code># Read and use the bdtnp geometry
#locations = np.loadtxt(datafile_geom, usecols=range(3), skiprows=1)
locations = novosparc.io.load_target_space(datafile_geom, is_2D=True)
locations = locations[:3039, :] # we are just focusing on the first 3039 positions
print(&quot;Number of location in the tissue &quot;,locations.shape[0])
print(locations[0:5,:])</code></pre>
<p>Each line in the geometry file gives you the XYZ coordinate of a cell.</p>
<pre class="python3"><code>plt.figure()
plt.scatter(locations[:, 0], locations[:, 1], s=1)
plt.show()
plt.close()
</code></pre>
<p>We can visualize all the position for which <span class="math inline">\(Z=1\)</span></p>
<pre class="python3"><code>plt.figure()
plt.scatter(locations[:, 0], locations[:, 1], s=1)
plt.show()
plt.close()
</code></pre>
</div>
<div id="alternative-tissue-spaces" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Alternative tissue spaces</h3>
<p>If you do not know the cellular locations of your tissue, you can create them. novoSpaRc has two functions that enable to construct them quickly.
#### case 2: you know the shape of your tissue
he function <em>create_target_space_from_image</em> creates a target space by using a user-generated image. The image has to be a black shape drawn onto a white background. You can do it in Fiji for instance.</p>
<pre class="python3"><code>tissue_path = &#39;data/data_day2/tissue_example.png&#39;
locations2   = novosparc.geometry.create_target_space_from_image(tissue_path)
print(locations2[:3,])</code></pre>
<p>This creates a very large number of locations, so it is better to subsample them.</p>
<pre class="python3"><code>print (&#39;dimension of original tissue space:&#39;, locations2.shape)
locations2     = locations2[np.random.choice(locations2.shape[0], 2000), :]
print (&#39;dimension of downsampled tissue space:&#39;, locations2.shape[0])
plt.figure()
plt.scatter(locations2[:, 0], locations2[:, 1],s=1)
plt.show()
plt.close()</code></pre>
<div id="case-3-you-assume-that-your-tissue-is-a-rectangle" class="section level4">
<h4><span class="header-section-number">3.2.2.1</span> case 3 : you assume that your tissue is a rectangle</h4>
<p>The function <em>construct_target_grid</em> can be used to construct a basic rectangular grid as target space. The only parameter is the total number of cellular locations on the target grid.</p>
<pre class="python3"><code>locations3 = novosparc.geometry.construct_target_grid(500)
plt.figure()
plt.scatter(locations3[:, 0], locations3[:, 1], s=1)
plt.show()
plt.close()</code></pre>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="basic-vizualization.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="predict-the-spatial-transcriptome.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
